src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -422,11 +431,33 @@ private TestResult ExecuteInternal(object?[]? arguments, ExecutionContext? execu
                     if (RunTestInitializeMethod(_classInstance, result, ref executionContext, timeoutTokenSource))
                     {
                         hasTestInitializePassed = true;
-                        FixtureMethodRunner.RunOnContext(executionContext, () =>
+                        var tcs = new TaskCompletionSource<object?>();
+#pragma warning disable VSTHRD101 // Avoid unsupported async delegates
+                        FixtureMethodRunner.RunOnContext(executionContext, async () =>
```

Comment:
TODO: Refactor to `RunOnContextAsync` and move these helper out of `FixtureMethodRunner` to a new helper `ExecutionContextHelpers`